gordo_components/builder/build_model.py

```diff
@@ -75,16 +77,37 @@ def build_model(
 
     scores: Dict[str, Any]
     if hasattr(model, "score"):
-        cv_scores = cross_val_score(model, X, y, cv=TimeSeriesSplit(n_splits=3))
+        n_splits = 3
+        cv = cross_validate(
+            model, X, y, return_estimator=True, cv=TimeSeriesSplit(n_splits=n_splits)
+        )
+
         scores = {
             "explained-variance": {
-                "mean": cv_scores.mean(),
-                "std": cv_scores.std(),
-                "max": cv_scores.max(),
-                "min": cv_scores.min(),
-                "raw-scores": cv_scores.tolist(),
+                "mean": cv["test_score"].mean(),
+                "std": cv["test_score"].std(),
+                "max": cv["test_score"].max(),
+                "min": cv["test_score"].min(),
+                "raw-scores": cv["test_score"].tolist(),
             }
         }
+
+        # adding loss function to scores if Gordo model is a KerasBaseEstimator
+        gordo_model = _get_final_gordo_base_step(model)
+        if issubclass(gordo_model.__class__, KerasBaseEstimator):
+            val_losses = np.array([])
+            for mod in cv["estimator"]:
```

Comment:
So this loops over each of the trained models? You want the loss for all of them? And does it make sense to have the mean/std, etc? If those values are not meaningful for anything really then it is better to not add them.

Reply:
i think it makes sense to have the mean/std/min/max especially if we decide to increase the number of splits as it would give us an idea of the variability of the validation losses for the different splits.